src/core/public/core_system.ts

```diff
@@ -168,12 +179,27 @@ export class CoreSystem {
     });
 
     const timing = this.getLoadMarksInfo();
+
+    const navigatorExt = navigator as ExtendedNavigator;
+    const navigatorInfo: Record<string, string> = {};
+    if (navigatorExt.connection?.effectiveType) {
+      navigatorInfo.effectiveNetworkType = navigatorExt.connection.effectiveType;
+    }
+    if (navigatorExt.deviceMemory) {
+      navigatorInfo.deviceMemory = String(navigatorExt.deviceMemory);
+    }
+    if (navigatorExt.hardwareConcurrency) {
+      navigatorInfo.hardwareConcurrency = String(navigatorExt.hardwareConcurrency);
+    }
```

Comment:
nit: I'd move this to a separate method (potentially in another file), similar to `fetchOptionalMemoryInfo` so it's easier to test and keeps the `core_system.ts` file a bit tighter (it's already a huge file and readability is kind of hard, IMO).